### ISSUE TITLE
Use case where newline-per-chained-call doesn't work well

### ISSUE THREAD

**Issue by Aequitas737** [eslint/eslint#12577]
**What rule do you want to change?**
newline-per-chained-call

**Does this change cause the rule to produce more or fewer warnings?**
Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
My guess would be an option for it

**Please provide some example code that this change will affect:**

```js
when(fn)
            .mockReturnValue(mockSync)
            .calledWith('param1').mockReturnValue(specificMockSync)
            .calledWith('param2').mockReturnValue(anotherSpecificMockSync)
            .calledWith('param3').mockReturnValue(yetAnotherSpecificMockSync);
```

According to the rule the code should be like this:
```js
when(fn)
            .mockReturnValue(mockSync)
            .calledWith('param1').mockReturnValue(specificMockSync)
            .calledWith('param2')
            .mockReturnValue(anotherSpecificMockSync)
            .calledWith('param3')
            .mockReturnValue(yetAnotherSpecificMockSync);
```
Which doesn't make logical sense, since the calledWith and mockReturnValue are linked (logically).

**What does the rule currently do for this code?**
Error for the last and second last `mockReturnValue`

**What will the rule do after it's changed?**
No error

**Are you willing to submit a pull request to implement this change?**
I'm not sure how this will be implemented. But yes I would if I had some guidance/feedback.

What does everyone think/suggest for this use-case?

**Comment by kaicataldo** [comment]
Thanks for the proposal. What kind of heuristics do you think ESLint should use to determine what should be allowed on the same line?

**Comment by Aequitas737** [comment]
Not really sure. 

Maybe something to check if the lines follow a pattern?

**Comment by kaicataldo** [comment]
Can you explain how it would do this? I don't see how ESLint can know what methods can be inlined without being explicitly given the method names that are allowed next to each other. 

I actually think the best thing to do in this case would be to use an `eslint-disable-*` comment.

**Comment by Aequitas737** [comment]
I've got no idea of the inner workings of eslint. I was thinking of using [some regex](https://regex101.com/r/V8we5D/1) to get the name and then as long as it's used previously it should be allowed.

However this wouldn't handle the below case:

```
when(fn)
            .mockReturnValue(mockSync)
            .mockReturnValue(mockSync) // just a contrived example
            .mockReturnValue(mockSync)
            .calledWith('param3').mockReturnValue(yetAnotherSpecificMockSync);
```

So yeh I'm not really sure there's a possible solution. I'll be disabling this rule most likely.
